Sources/Core/Client.swift

```diff
@@ -314,8 +321,14 @@ public actor Client {
             let pack = try Converter.fromChangePack(result.changePack)
             try await doc.applyChangePack(pack: pack)
 
-            self.attachmentMap[doc.getKey()] = Attachment(doc: doc, isRealtimeSync: isRealtimeSync, peerPresenceMap: [String: PresenceInfo](), remoteChangeEventReceived: false)
-            self.runWatchLoop()
+            if await doc.status == .removed {
+                return doc
+            }
```

Comment:
This is about what we briefly talked about this morning. 
When applying the changePack received in the attach, if the document is in the "removed" state, should we simply return the doc? Or would it be better to consider it as attach failure?